src/middlewared/middlewared/plugins/pool.py

```diff
@@ -1590,11 +1602,19 @@ def __find_disk_from_topology(self, label, pool):
             verrors.add('id', 'Pool already locked.')
 
         if not verrors:
-            if not await self.middleware.call('disk.geli_testkey', pool, passphrase):
+            # Make sure that this pool is not being used by system dataset service
+            if pool['name'] == (await self.middleware.call('systemdataset.config'))['pool']:
                 verrors.add(
-                    'passphrase',
-                    'Please provide a valid passphrase to lock the pool'
+                    'id',
+                    'Pool contains the system dataset and cannot be locked. Please select a different pool or '
+                    'configure the system dataset to be on a different pool'
                 )
+            else:
+                if not await self.middleware.call('disk.geli_testkey', pool, passphrase):
+                    verrors.add(
+                        'passphrase',
+                        'Please provide a valid passphrase to lock the pool'
```

Comment:
```suggestion
                        'Enter the passphrase to lock the pool.'
```

Reply:
The users in this case have already entered a passphrase, it's just not valid. Should I add `valid` like `Enter a valid passphrase to lock the pool.` ?

Reply:
The best way is to be explicit.
```suggestion
                        'The entered passphrase was not valid. Please enter the correct passphrase to lock the pool.'
```